From fd985189291d8562afb9541cd0d11ffe03de544e Mon Sep 17 00:00:00 2001
From: brokestar233 <3765589194@qq.com>
Date: Sun, 17 Aug 2025 19:11:10 +0800
Subject: [PATCH] mm: uksm: Lower uksm_abundant_threshold to 5

Reduce the uksm_abundant_threshold from 10 to 5 to make the deduplication
process more sensitive to memory pages with lower deduplication ratios.
This change aims to improve memory efficiency by allowing UKSM to consider
pages with a lower deduplication ratio as abundant, potentially increasing
the frequency of deduplication in certain workloads.

Signed-off-by: brokestar233 <3765589194@qq.com>
---
 mm/uksm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: common/mm/uksm.c
===================================================================
--- common/mm/uksm.c	2025-11-01 01:23:11.899281991 +1030
+++ common/mm/uksm.c	2025-11-01 01:23:11.895450485 +1030
@@ -560,7 +560,7 @@
 static unsigned int uksm_thrash_threshold = 50;
 
 /* How much dedup ratio is considered to be abundant*/
-static unsigned int uksm_abundant_threshold = 10;
+static unsigned int uksm_abundant_threshold = 5;
 
 /* All slots having merged pages in this eval round. */
 struct list_head vma_slot_dedup = LIST_HEAD_INIT(vma_slot_dedup);
@@ -5412,7 +5412,7 @@
 	addr1 = kmap_atomic(p1);
 	addr2 = kmap_atomic(p2);
 	memset(addr1, get_random_u32(), PAGE_SIZE);
-	memcpy(addr2, addr1, PAGE_SIZE);
+	copy_page(addr2, addr1);
 
 	/* make sure that the two pages differ in last byte */
 	addr2[PAGE_SIZE-1] = ~addr2[PAGE_SIZE-1];
